cmd/topology/topology_test.go

```diff
@@ -0,0 +1,248 @@
+	defer closer()
+	tWithConfig := &topopb.Topology{
+		Nodes: []*topopb.Node{{
+			Name: "resettable1",
+			Type: topopb.Node_Type(1001),
+			Config: &topopb.Config{
+				ConfigData: &topopb.Config_Data{
+					Data: []byte("somebytes"),
+				},
+			},
+		}, {
+			Name: "resettable2",
+			Type: topopb.Node_Type(1001),
+			Config: &topopb.Config{
+				ConfigData: &topopb.Config_File{
+					File: confFile.Name(),
+				},
+			},
+		}, {
+			Name: "notresettable1",
+			Type: topopb.Node_Type(1002),
+		}},
+	}
+	fConfig, closer := writeTopology(t, tWithConfig)
+	defer closer()
+	tWithConfigRelative := &topopb.Topology{
+		Nodes: []*topopb.Node{{
+			Name: "resettable1",
+			Type: topopb.Node_Type(1001),
+			Config: &topopb.Config{
+				ConfigData: &topopb.Config_Data{
+					Data: []byte("somebytes"),
+				},
+			},
+		}, {
+			Name: "resettable2",
+			Type: topopb.Node_Type(1001),
+			Config: &topopb.Config{
+				ConfigData: &topopb.Config_File{
+					File: filepath.Base(confFile.Name()),
+				},
+			},
+		}, {
+			Name: "notresettable1",
+			Type: topopb.Node_Type(1002),
+		}},
+	}
+	fConfigRelative, closer := writeTopology(t, tWithConfigRelative)
+	defer closer()
+	tWithConfigDNE := &topopb.Topology{
+		Nodes: []*topopb.Node{{
+			Name: "resettable1",
+			Type: topopb.Node_Type(1001),
+			Config: &topopb.Config{
+				ConfigData: &topopb.Config_Data{
+					Data: []byte("somebytes"),
+				},
+			},
+		}, {
+			Name: "resettable2",
+			Type: topopb.Node_Type(1001),
+			Config: &topopb.Config{
+				ConfigData: &topopb.Config_File{
+					File: "dne",
+				},
+			},
+		}, {
+			Name: "notresettable1",
+			Type: topopb.Node_Type(1002),
+		}},
+	}
+	fConfigDNE, closer := writeTopology(t, tWithConfigDNE)
+	defer closer()
+	node.Register(topopb.Node_Type(1001), NewR)
+	node.Register(topopb.Node_Type(1002), NewNR)
+	tests := []struct {
+		desc    string
+		args    []string
+		tFile   string
+		wantErr string
+	}{{
+		desc:    "no args",
+		wantErr: "invalid args",
+		args:    []string{"reset"},
+	}, {
+		desc:    "no file",
+		args:    []string{"reset", "filedne"},
+		wantErr: "no such file",
+	}, {
+		desc:    "valid topology no skip",
+		args:    []string{"reset", fNoConfig.Name(), "--skip=false"},
+		wantErr: "node notresettable1 is not resettable",
+	}, {
+		desc: "valid topology no skip",
+		args: []string{"reset", fNoConfig.Name(), "--skip"},
+	}, {
+		desc: "valid topology no skip nothing to push",
+		args: []string{"reset", fNoConfig.Name(), "--skip", "--push"},
+	}, {
+		desc: "valid topology no skip config to push",
+		args: []string{"reset", fConfig.Name(), "--skip", "--push"},
+	}, {
+		desc: "valid topology push with relative file location",
+		args: []string{"reset", fConfigRelative.Name(), "--skip", "--push"},
+	}, {
+		desc:    "valid topology push with config DNE",
+		args:    []string{"reset", fConfigDNE.Name(), "--skip", "--push"},
+		wantErr: "no such file or directory",
+	}}
+
+	rCmd := New()
+	var s string
```

Comment:
Where is this being used?

Reply:
hmm not sure why lint didn't find that but good catch

Reply:
it was actgually used for holding the persistent flag but found a better method